src/tests/end2end/CopyTests.cpp

```diff
@@ -87,51 +88,63 @@ class CopyTests_T2B : public CopyTests {
             uint32_t height = textureSpec.height >> textureSpec.level;
             uint32_t rowPitch = Align(kBytesPerTexel * width, kTextureRowPitchAlignment);
             uint32_t texelsPerRow = rowPitch / kBytesPerTexel;
-            uint32_t texelCount = texelsPerRow * (height - 1) + width;
-
-            // Create an upload buffer and use it to populate the `level` mip of the texture
-            std::vector<RGBA8> textureData(texelCount);
-            FillTextureData(width, height, rowPitch / kBytesPerTexel, textureData.data());
-            dawn::Buffer uploadBuffer = utils::CreateBufferFromData(device, textureData.data(), static_cast<uint32_t>(sizeof(RGBA8) * textureData.size()), dawn::BufferUsageBit::TransferSrc);
-
-            dawn::CommandBuffer commands[2];
-
-            commands[0] = device.CreateCommandBufferBuilder()
-                .CopyBufferToTexture(uploadBuffer, 0, rowPitch, texture, 0, 0, 0, width, height, 1, textureSpec.level)
-                .GetResult();
+            uint32_t texelCountPerLayer = texelsPerRow * (height - 1) + width;
+
+            // Create an upload buffer and use it to populate all slices of the `level` mip of the texture
+            std::vector<std::vector<RGBA8>> textureArrayData(textureSpec.arrayLayer);
+            std::vector<dawn::CommandBuffer> commands(textureSpec.arrayLayer * 2);
```

Comment:
We can keep only one command buffer and have multiple copies in it.

Reply:
Done